packages/@aws-cdk/app-staging-synthesizer/README.md

```diff
@@ -0,0 +1,291 @@
+> announced in the release notes. This means that while you may use them, you may need to update
+> your source code when upgrading to a newer version of this package.
+
+---
+
+<!--END STABILITY BANNER-->
+
+This library includes constructs aimed at replacing the current model of bootstrapping and providing
+greater control of the bootstrap experience to the CDK user. The important constructs in this library
+are as follows:
+
+- the `IStagingResources` interface: a framework for an app-level bootstrap stack that handles
+  file assets and docker assets.
+- the `DefaultStagingStack`, which is a works-out-of-the-box implementation of the `IStagingResources`
+  interface.
+- the `AppStagingSynthesizer`, a new CDK synthesizer that will synthesize CDK applications with
+  the staging resources provided.
+
+## Bootstrap Model
+
+Our current bootstrap model looks like this, when you run `cdk bootstrap aws://<account>/<region>` :
+
+```mermaid
+graph TD
+    A(Bootstrap Stack) --- B(CloudFormationExecutionRole <br> DeploymentActionRole <br> LookupRole <br> FilePublishingRole <br> ImagePublishingRole <br> StagingBucket <br> ContainerAssetsRepository <br> FileAssetsBucketEncryptionKey)
+```
+
+Your CDK Applicaiton utilizes some of these resources when deploying. For example, if you have a file asset,
+it gets uploaded to the `StagingBucket` using the `FilePublishingRole` when you run `cdk deploy`.
+
+This library introduces an alternate model to bootstrapping, by splitting out essential CloudFormation iam roles
+and staging resources. There will still be a Bootstrap Stack, but this will only contain IAM roles necessary for
+CloudFormation deployment. Each CDK App will instead be in charge of it's own staging resources, including the
+S3 Bucket, ECR Repositories, and associated IAM roles. It works like this:
+
+The Staging Stack will contain, on a per-need basis, 
+
+- 1 S3 Bucket with KMS encryption for all file assets in the CDK App.
+- An ECR Repository _per_ image (and its revisions).
+- IAM roles with access to the Bucket and Repositories.
+
+```mermaid
+graph TD
+    A(Bootstrap Stack) --- B(CloudFormationExecutionRole <br> DeploymentActionRole <br> LookupRole)
+    C(CDK App with File Asset) --- D(Staging Stack A) 
+    C --- I(Stack A)
+    I --- J(File Asset)
+    F(CDK App with File/Image Asset) --- G(Staging Stack B)
+    F --- K(Stack B)
+    K --- L(File Asset <br> Image Asset)
+    G --- H(FilePublishingRole <br> ImagePublishingRole <br> StagingBucket <br> ContainerAssetsRepository <br> FileAssetsBucketEncryptionKey)
+    D --- E(FilePublishingRole <br> StagingBucket <br> FileAssetsBucketEncryptionKey)
+    M(CDK App with no Assets) --- N(Stack C)
+    N --- O(<br>)
+```
+
+This allows staging resources to be created when needed next to the CDK App. It has the following
+benefits:
+
+- Bootstrapping will be faster since the heavy resource of a KMS key is no longer involved.
```

Comment:
Not really a big benefit, and in any case the KMS key is just moved so now a different step will be slower.